File: frontend/lib/models/job.dart
class Job {
  final int id;
  final String title;
  final String company;
  final String? location;
  final String? description;
  final String? salary;
  final String? source;
  final String? applyUrl;
  final DateTime? createdAt;
  final double? matchScore;

  Job({
    required this.id,
    required this.title,
    required this.company,
    this.location,
    this.description,
    this.salary,
    this.source,
    this.applyUrl,
    this.createdAt,
    this.matchScore,
  });

  factory Job.fromJson(Map<String, dynamic> json) {
    return Job(
      id: json['id'] as int,
      title: json['title'] as String,
      company: json['company'] as String,
      location: json['location'] as String?,
      description: json['description'] as String?,
      salary: json['salary'] as String?,
      source: json['source'] as String?,
      applyUrl: json['apply_url'] as String?,
      createdAt: json['created_at'] != null ? DateTime.tryParse(json['created_at']) : null,
      matchScore: (json['match_score'] as num?)?.toDouble(),
    );
  }
}

File: backend/app/models/job.py
from datetime import datetime

from sqlalchemy import DateTime, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class Job(Base):
    __tablename__ = "jobs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    title: Mapped[str] = mapped_column(String(255), nullable=False, index=True)
    company: Mapped[str] = mapped_column(String(255), nullable=False, index=True)
    location: Mapped[str | None] = mapped_column(String(255), nullable=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    salary: Mapped[str | None] = mapped_column(String(100), nullable=True)
    source: Mapped[str | None] = mapped_column(String(100), nullable=True)
    apply_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    posted_date: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )

    applications = relationship("Application", back_populates="job", cascade="all,delete")
